proposal.rst

```diff
@@ -197,45 +233,88 @@ Project Details
 
 **Describe any existing work and concepts on which your project is based.**
 
-FILLME
+
+I am still trying to see if we can utilize the N-gram algorithm to help on this task.I made a google `doc <https://docs.google.com/document/d/1yXdnAhtNKWcODLLBy2hB2pluQ2nr3EfZJP2_F1XIpDs/edit
+/>`_ that I keep updated with results I reached so far.
+
 
 **Do you have any preliminary findings or results which suggest that your
 approach is possible and likely to succeed?**
 
-FILLME
+There's approach addressed in a `patch <https://oligarchy.co.uk/xapian/patches/docid-ranges-in-matcher.patch/>`_ by Olly,
+that make use of the fact that we have the ids for the first and last documents now.
+Example: for the "MultiAndPostList", we would mutiply the probability of each term independantly, and we would do that by
+divide the term frequnecy(from term_freq_est() function) by the database size. But now we can get the range from the first docid to the last docid
+(last - first +1), and we use that range instead of the whole database size.
+
+As for now, I am trying to come by a better approach for this task.
 
 **What other approaches have you considered, and why did you reject those in
 favour of your chosen approach?**
 
-FILLME
+
+To store the count of term pairs in the database, but this can pretty huge
+(ie 10k terms would be (10k C 2) entries which around 50 million).
+This would give us an accurate number of the results, but it can grow exponentially
+and take huge space.
+
 
 **Please note any uncertainties or aspects which depend on further research or
 investigation.**
 
-FILLME
+I am still not sure if we can come up with better approach. Or how better the result would be for approach using the first and last doc ids
 
 **How useful will your results be when not everything works out exactly as
 planned?**
 
-FILLME
+It would still produce better estimates as we reduced the number of documents we divide by.
 
 Project Timeline
 ----------------
 
+* Three Weeks of Bonding:
+
+ 	* Get more familiar with the code base, especially the Matcher code.
+
+ 	* Build some tests on debug mode to see how functions are invoked.
+
+ 	* Search more for ways to tackle this task.
+
+* Two Weeks: Working on OP_AND, OP_OR, and OP_AND_NOT (Code, testing, and doumentation)
+
+
+* One and half Week: Working on OP_XOR,and OP_AND_MAYBE (Code, testing, and doumentation.)
+
+
+10 June : Final exams (this is an estimate,official schedule isn't announced yet.)
+
+
+* Two Weeks: Working on OP_FILTER, OP_NEAR, and OP_PHRASE (Code, testing, and doumentation.)
+
+* Two Weeks: Working on OP_VALUE_RANGE, OP_VALUE_LE, OP_VALUE_LT, OP_VALUE_GE, and OP_VAUE_GT (Code, testing, and doumentation).
+
+* Two Weeks: Working on OP_SCALE_WEIGHT,and OP_ELITE_LIST (Code, testing, and doumentation.)
```

Comment:
You haven't followed our guidance about breaking things down to a week _at most_. Additionally, I can't tell from your proposal at the moment anything concrete about what you'll be doing in any of these periods. I'd like to see a discussion of the background to the approach overall, so I understand what work is needed specifically for OP_FILTER, for instance.

In particular, we suggest in our guidance that your project should be made up of several smaller projects, each of which can be completed and merged and which will produce some improvement for Xapian. I don't understand at the moment where this work could be merged, or what it would do. For instance, if we merged it at this point, what are we going to get next out of the work on OP_SYNONYM and OP_MAX?